src/Microsoft.Framework.PackageManager/Install/AppCommandsFolderRepository.cs

```diff
@@ -0,0 +1,141 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Framework.Runtime;
+using NuGet;
+
+namespace Microsoft.Framework.PackageManager
+{
+    internal class AppCommandsFolderRepository : IAppCommandsRepository
+    {
+        private readonly NuGet.IFileSystem _commandsFolder;
+
+        // Key = command; Value = app name
+        private IDictionary<string, NuGet.PackageInfo> _commands = new Dictionary<string, NuGet.PackageInfo>();
+
+        public AppCommandsFolderRepository(string commandsFolder)
+        {
+            _commandsFolder = new NuGet.PhysicalFileSystem(commandsFolder);
+        }
+
+        public IFileSystem Root
+        {
+            get
+            {
+                return _commandsFolder;
+            }
+        }
+
+        public IFileSystem PackagesRoot
+        {
+            get
+            {
+                return _commandsFolder.GetDirectory(InstallGlobalCommand.TargetPackagesFolderName);
+            }
+        }
+
+        public IEnumerable<string> Commands
+        {
+            get
+            {
+                return _commands.Keys;
+            }
+        }
+
+        public NuGet.PackageInfo FindCommandOwner(string command)
+        {
+            NuGet.PackageInfo appPackage;
+            _commands.TryGetValue(command, out appPackage);
+            return appPackage;
+        }
+
+        public void Remove(string commandName)
+        {
+            string commandFileName = commandName +
+                (PlatformHelper.IsWindows ? ".cmd" : string.Empty);
+
+            _commands.Remove(commandName);
+            _commandsFolder.DeleteFile(commandFileName);
+        }
+
+        public void Load()
+        {
+            _commands = new Dictionary<string, NuGet.PackageInfo>();
+
+            string pathFilter = PlatformHelper.IsWindows ? "*.cmd" : "*.*";
```

Comment:
Yup, starting a thread with the CLR team because they don't expose the OS version on CoreCLR